cli/src/build/onboarding/ui/app.tsx

```
@@ -86,6 +91,53 @@ async function runRunnerCommand(runner: string, args: string[]): Promise<{ succe
+ *   3. "Type a custom command" escape hatch
+ *   4. "Skip build step" escape hatch (for raw HTML Capacitor apps)
+ *
+ * Showing ALL scripts (not just "build"-ish ones) matches what the user
+ * asked for: pick from package.json, never auto-guess. Filtering risks
+ * hiding the exotic name a user actually wants.
+ */
+function buildScriptPickerOptions(scripts: Record<string, string>, recommended: string | null): BuildScriptOption[] {
+  const options: BuildScriptOption[] = []
+  const seen = new Set<string>()
+
+  if (recommended && Object.prototype.hasOwnProperty.call(scripts, recommended)) {
```

ESLint

[error] Use Object.hasOwn() instead of hasOwnProperty

(e18e/prefer-object-has-own)